### ISSUE TITLE
Something wrong with updatePlanes

### ISSUE THREAD

**Issue by aferriss** [sortofsleepy/arkit-tests#14]
Hey I just pulled from master and noticed that a few seconds after running my app it begins to crash. I did a stack trace and it seems like the error is coming from pushing new planes into the planes vector. Logging the size of that vector shows that it quickly reaches into the hundreds of thousands or millions...which would explain the crash.

I did some poking around but I couldn't seem to pinpoint the specific commit where this changed. I switched back over to develop and this doesn't happen, so maybe whatever changed between them will fix.

![screen shot 2017-09-13 at 3 18 26 pm](https://user-images.githubusercontent.com/3698659/30404158-0ff60f92-989a-11e7-9bc4-e8d22710342a.png)
![screen shot 2017-09-13 at 3 38 56 pm](https://user-images.githubusercontent.com/3698659/30404159-1206dd84-989a-11e7-97a2-f02df19d5090.png)

**Comment by sortofsleepy** [comment]
@aferriss ah crud - thanks for catching that.  Apologies if the commit history was a bit hard to follow, I've been trying to make changes strictly on develop but on occasion I slip up and forget this isn't a personal project anymore haha. 

I ran into this too a couple days ago as well and didn't realize the bad code made it into master. Turns out I didn't think through my for loops the first time around doh* 😅 

I've just gone ahead and updated AnchorManager, feel free to test, I'll be testing when I get home as well.

**Comment by sortofsleepy** [comment]
ok! It appears to be working ok and nothing else has broken as far as I can tell. Gonna just close this now. feel free to re-open if it's not. 

Note - in case you tested the planes example I realized just now that I forgot to update the same example. I turned off auto updating for planes due to what I believe now was likely this very same issue. That's been updated too.

**Comment by aferriss** [comment]
Hey, something is still borked in here.

If you set shouldUpdatePlanes to true it will immediately crash upon finding a plane. I think it has to do with this [line](https://github.com/sortofsleepy/ofxARKit/blob/master/src/ARAnchorManager.mm#L185). 

I logged the distance value and it was some crazy high number, so, I'm not sure it's returning the correct value. The crash happens when it goes to set a matrix on 

`planes[index].transform = paTransform;`


Removing that line fixed the issue for me (since you also have your increment in the loop named index).

**Comment by sortofsleepy** [comment]
@aferriss Good catch! I totally f'd up the first time around. Clearly I forgot to test that. Fixed in develop - will merge into master over the weekend. Other then that, any other issues? (I don't have a phone with me today + another project I gotta worry about)

**Comment by aferriss** [comment]
That's the only thing I've found, thanks for the quick fix!

**Comment by sortofsleepy** [comment]
closing again as I finally got around to moving fix into master after being sick for a bit.
